### PR TITLE
[LUM-409] Implement Cosmos v0.46.0

### PR DESCRIPTION
**Pull request by Segfaultd**
### Introduction

This PR introduces a few changes related to the used Cosmos SDK version, but also the used IBC version.

It bumps the SDK to 0.46.0, which is a major upgrade with breaking changes:

- Store keys now lives under the storetypes and not under the sdk namespace anymore
- The mint module now takes an additional parameter that we don't require at the moment
- The legacy REST routes are fully deprecated
- The IBC Fee keeper is now required in order to initialize the IBC module
- The default keyring backend is now "file" and not "testing" anymore
- We now use iso-production parameters in unit testing, especially when it comes to the default Bond Denom
- A single Entry point for testing is now provided, it ensures the app is fully initialized with the correct parameters, voting power and validator set

It also bumps IBC-Go to v5.

Upgrade guides

https://github.com/cosmos/ibc-go/blob/main/docs/migrations/v3-to-v4.md

https://github.com/cosmos/ibc-go/blob/main/docs/migrations/v4-to-v5.md

### Testing

The unit tests have been patched to compile but no logic was touched.
Considering the deep changes we introduce, we must carefully test this on a dedicated network before sending to mainnet.

### CODE REVIEW

File: x/dfract/client/cli/proposal.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/cosmos/cosmos-sdk/client/tx"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/gov/client/cli"
-	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govv1betatypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
```
Comment (lebascou) [inline x/dfract/client/cli/proposal.go:10]: I believe this should be `govtypesv1beta1` for consistency purpose.

**Review by lebascou — COMMENTED** [review]
Implementation LGTM, we need to thoroughly test the following now:
- Keplr: RPC behaviour since it's been heavily deprecated and edited
- IBC: All features in a production like environment

File: app/app.go [hunk 2]
```diff
@@ -634,6 +644,11 @@ func (app *App) registerUpgradeHandlers() {
 		return app.mm.RunMigrations(ctx, app.configurator, fromVM)
 	})
 
+	app.UpgradeKeeper.SetUpgradeHandler("v1.3.0", func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		app.Logger().Info("v1.3.0 upgrade applied")
```
[the remark this answers is not in the document]
Comment (Segfaultd) [reply app/app.go:648]: Well technically everything is running in that lambda.

File: app/keepers.go [hunk 3]
```diff
@@ -190,6 +194,14 @@ func (app *App) InitNormalKeepers() {
 		appCodec, keys[ibchost.StoreKey], app.GetSubspace(ibchost.ModuleName), app.StakingKeeper, app.UpgradeKeeper, app.ScopedIBCKeeper,
 	)
 
+	// Initialize the IBC Fee keeper
+	app.IBCFeeKeeper = ibcfeekeeper.NewKeeper(
+		appCodec, app.keys[ibcfeetypes.StoreKey], app.GetSubspace(ibcfeetypes.ModuleName),
+		app.IBCKeeper.ChannelKeeper, // may be replaced with IBC middleware
```
[the remark this answers is not in the document]
Comment (Segfaultd) [reply app/keepers.go:200]: May be replaced in the future if we require an IBC fee middleware.

File: app/app.go [hunk 4]
```diff
@@ -664,4 +679,8 @@ func (app *App) registerUpgradeHandlers() {
 		storeUpgrades := storetypes.StoreUpgrades{}
 		app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
 	}
-}
```
Comment (lebascou) [inline app/app.go:667]: Looks like a missing bracket 💡

**Review by lebascou — CHANGES_REQUESTED** [review]

**Review by lebascou — APPROVED** [review]

**Pull request merged**